src/plugins/mf_priority.cpp

```diff
@@ -807,6 +851,44 @@ static int depend_cb (flux_plugin_t *p,
 }
 
 
+/*
+ * When a job gets to the job.state.sched state, pack a timestamp to be
+ * associated with the job in the case that it is eligible to be run but
+ * cannot (e.g due to resource constraints). The timestamp will be used
+ * to calculate the "age" of a job and increase its priority according to
+ * the time it has been waiting to run.
+ */
+static int sched_cb (flux_plugin_t *p,
+                     const char *topic,
+                     flux_plugin_arg_t *args,
+                     void *data)
+{
+    double t_released = 0.0;
+    int urgency = 0;
+    long int jobid;
+    flux_t *h = flux_jobtap_get_flux (p);
+
+    // unpack urgency
+    if (flux_plugin_arg_unpack (args,
+                                FLUX_PLUGIN_ARG_IN,
+                                "{s:I, s:i}",
+                                "id", &jobid,
+                                "urgency", &urgency) < 0)
+        return flux_jobtap_error (p, args, "unable to unpack urgency");
+
+    if (urgency > 0) {
+        // the job is not being held, and therefore we need to get the current
+        // time to keep track of how long the job has been released and waiting
+        t_released = std::chrono::duration_cast <std::chrono::duration<double>>
+            (std::chrono::system_clock::now ().time_since_epoch ()).count ();
+        // store job id, t_released in map
+        released_jobs[jobid] = t_released;
+    }
```

Comment:
Question: If urgency is 0, should `released_jobs.erase (jobid)` be called?

Reply:
It seems like there may be times when the `t_released` for a job should _not_ be updated in SCHED state. E.g. if a priority adjustment is made and the job was not previously held, shouldn't the age stay the same?